app/src/main/java/nerd/tuxmobil/fahrplan/congress/preferences/AlarmTonePreference.kt

```diff
@@ -0,0 +1,84 @@
+package nerd.tuxmobil.fahrplan.congress.preferences
+
+import android.content.Context
+import android.content.Intent
+import android.media.RingtoneManager
+import android.net.Uri
+import android.provider.Settings
+import android.util.AttributeSet
+import androidx.fragment.app.Fragment
+import androidx.preference.Preference
+import nerd.tuxmobil.fahrplan.congress.extensions.withExtras
+import nerd.tuxmobil.fahrplan.congress.utils.AlarmToneConversion
+
+/**
+ * Represents an alarm tone preference.
+ */
+class AlarmTonePreference : Preference {
+
+    companion object {
+
+        /**
+         * See [Settings.System.DEFAULT_ALARM_ALERT_URI]
+         */
+        val DEFAULT_VALUE_URI: Uri = Settings.System.DEFAULT_ALARM_ALERT_URI
+
+        /**
+         * Content `Uri` string representation of [DEFAULT_VALUE_URI].
+         */
+        val DEFAULT_VALUE_STRING: String = DEFAULT_VALUE_URI.toString()
+
+    }
+
+    @Suppress("unused")
+    constructor(context: Context?) :
+            super(context) {
+        applyDefaultValue()
+    }
+
+    @Suppress("unused")
+    constructor(context: Context?, attrs: AttributeSet?) :
+            super(context, attrs) {
+        applyDefaultValue()
+    }
+
+    @Suppress("unused")
+    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) :
+            super(context, attrs, defStyleAttr) {
+        applyDefaultValue()
+    }
+
+    @Suppress("unused")
+    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
+            super(context, attrs, defStyleAttr, defStyleRes) {
+        applyDefaultValue()
+    }
+
+    private fun applyDefaultValue() {
+        setDefaultValue(DEFAULT_VALUE_STRING)
+    }
+
+    override fun onClick() {
+        super.onClick()
```

Comment:
This does nothing (and you don't really want the super class to do anything if you're handling the click yourself).

Reply:
:heavy_check_mark: Removed in d350abd490dc6c769b9b0cd9b4c512ad209fd314.